File: Python/EstruturaDeRepeticao/exercicio_11.py
"""
Faça um programa que receba dois números inteiros e gere os números inteiros que estão no intervalo compreendido por eles.

Altere o programa anterior para mostrar no final a soma dos números.
"""

numero1 = int(input(f"Digite o 1º número: "))
numero2 = int(input(f"Digite o 2º número: "))
soma = 0

print(f"Os números inteiros entre {numero1} e {numero2} são:", end=" ")

if numero1 > numero2:
    for i in range(numero1-1, numero2, -1):
        print(i, end=" ")
        soma += i
else:
    for i in range(numero1, numero2):
        print(i, end=" ")
        soma += i

print(f"\nA soma dos números é: {soma}")

File: Java/EstruturaDeRepeticao/Exercicio11.java
// Faça um programa que receba dois números inteiros e gere os números inteiros que estão no intervalo compreendido por eles.
// Altere o programa anterior para mostrar no final a soma dos números.
package EstruturaDeRepeticao;
import java.util.Scanner;

public class Exercicio11 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite o primeiro número inteiro: ");
        int numero1 = scanner.nextInt();

        System.out.print("Digite o segundo número inteiro: ");
        int numero2 = scanner.nextInt();

        System.out.println("Números no intervalo entre " + numero1 + " e " + numero2 + ":");

        int soma = 0;

        if (numero1 <= numero2) {
            for (int i = numero1; i <= numero2; i++) {
                System.out.print(i + " ");
                soma += i;
            }
        } else {
            for (int i = numero1; i >= numero2; i--) {
                System.out.print(i + " ");
                soma += i;
            }
        }

        System.out.println("\nSoma dos números no intervalo: " + soma);

        scanner.close();
    }
}
